### PR TITLE
Fix running make from a testdir

### PR DESCRIPTION
**Pull request by adizaimi**
Currently, running make from a testdir is cumbersome and one has to provide testid and testrootdir.
But that is unnecessary and i am fixing in this PR. Now one can simply run make from a testdir, and it will automatically set testid and testrootdir. 
In order for those variables to be automatically set (by testcase.mk), a given test's Makefile will need to start with:
```sh
ifeq ($(TESTSROOTDIR),)
   include ../testcase.mk
else
   include $(TESTSROOTDIR)/testcase.mk
endif
```
See `basic.test` for an example.

### CODE REVIEW

File: tests/basic.test/Makefile [hunk 1]
```diff
@@ -1,2 +1,6 @@
+ifeq ($(TESTSROOTDIR),)
+  export TESTSROOTDIR=$(shell readlink -f $(PWD)/..)
```
Comment (mponomar) [inline tests/basic.test/Makefile]: Let's just go with ..?

File: tests/testcase.mk [hunk 2]
```diff
@@ -1,6 +1,9 @@
+ifeq ($(TESTSROOTDIR),)
```
Comment (mponomar) [inline tests/testcase.mk]: Maybe it's time we went through all the tests and changed it to include ../testcase.mk?

Comment (adizaimi) [inline tests/testcase.mk]: Ok, can do as as separate PR.
